### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by dariia-romanova**
[DEMO LINK](https://dariia-romanova.github.io/js_task_parse_list_DOM/)

### CODE REVIEW

**Review by SerhiiKirik — APPROVED** [review]
Good approach 🔥

File: src/scripts/main.js [hunk 1]
```diff
@@ -1,3 +1,50 @@
...
+  list.innerHTML = `
+  ${employees.map(employee => (`
+    <li
+      data-position = "${employee.dataset.position}"
+      data-salary = "${employee.dataset.salary}"
+      data-age = "${employee.dataset.age}"
+    >${employee.innerHTML.trim()}
+    </li>`
+  )).join('')}`;
+}
```
Comment (AlinaMatuschak) [inline src/scripts/main.js:35]: this function name is sortList. But there you also change dom. Maybe better rename function or change dom outside on the bottom of the file?

File: src/scripts/main.js [hunk 2]
```diff
@@ -1,3 +1,50 @@
...
+  const employees = [...children];
+
+  employees.sort((employee1, employee2) => {
+    const salary1 = employee1.dataset.salary;
+    const salary2 = employee2.dataset.salary;
+
+    return convertToNumber(salary2) - convertToNumber(salary1);
+  });
+
+  list.innerHTML = `
```
Comment (AlinaMatuschak) [inline src/scripts/main.js:26]: try use insertAdjacentHTML

File: src/scripts/main.js [hunk 3]
```diff
@@ -1,3 +1,50 @@
...
+function getEmployees(list) {
+  const children = list.children;
+  const employees = [...children];
+
+  return employees.map((employee) => ({
+    name: employee.innerHTML.trim(),
+    position: employee.dataset.position,
+    salary: employee.dataset.salary,
+    age: employee.dataset.age,
+  }));
```
Comment (AlinaMatuschak) [inline src/scripts/main.js:46]:
```suggestion
  return employees.map((employee) => ({
    name: employee.innerHTML.trim(),
    ...person.dataset,
  }));
```

File: src/scripts/main.js [hunk 4]
```diff
@@ -1,3 +1,50 @@
...
+
+    return true;
+  }).join('');
+}
+
+const listOfEmployees = document.querySelector('ul');
+
+function sortList(list) {
+  const children = list.children;
+  const employees = [...children];
```
Comment (AlinaMatuschak) [inline src/scripts/main.js:17]:
```suggestion
  const employees = [...list.children];
```

File: src/scripts/main.js [hunk 5]
```diff
@@ -1,3 +1,50 @@
...
+  return employees.map((employee) => ({
+    name: employee.innerHTML.trim(),
+    position: employee.dataset.position,
+    salary: employee.dataset.salary,
+    age: employee.dataset.age,
+  }));
+}
+
+sortList(listOfEmployees);
+getEmployees(listOfEmployees);
```
Comment (AlinaMatuschak) [inline src/scripts/main.js:50]: this function returns employees with names. But you don't use this value 🤔

**Review by AlinaMatuschak — CHANGES_REQUESTED** [review]
try to use employees obj with names to update dom

File: src/scripts/main.js (hunk 2 above)
Comment (dariia-romanova) [reply src/scripts/main.js]: But I need to replace html rather than add it. Can't I use innerHTML in this case?

Comment (AlinaMatuschak) [reply src/scripts/main.js]: You can leave default list

**Review by AlinaMatuschak — APPROVED** [review]
🔥

**Pull request closed without merging**